src/XML/CanonicalizableTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\XMLSecurity\XML;
+
+use DOMElement;
+use SimpleSAML\XMLSecurity\Utils\XML;
+
+/**
+ * A trait implementing the CanonicalizableInterface.
+ *
+ * @package simplesamlphp/xml-security
+ */
+trait CanonicalizableTrait
```

Comment:
Regarding the name of the trait and the interface....  Canonicalizable-what?  Maybe rename to CanonicalizableXMLTrait / CanonicalizableXMLInterface or CanonicalizableElementTrait / CanonicalizableElementInterface

Reply:
I saw this coming 😉 

Yeah, I already thought about `CanonicalizableElement*`, but since it was quite long I left it like that. I'll rename it 👍